cpp/test/Ice/custom/TestAMDI.cpp

```diff
@@ -4,18 +4,44 @@
 
 #include "TestAMDI.h"
 #include "Ice/Communicator.h"
+#include "TestHelper.h"
 
 using namespace std;
 using namespace Ice;
 using namespace Test;
 
+void
+TestIntfI::opShortArrayAsync(
+    pair<const int16_t*, const int16_t*> inSeq,
+    function<void(pair<const int16_t*, const int16_t*>, pair<const int16_t*, const int16_t*>)> response,
+    function<void(exception_ptr)>,
+    const Current&)
+{
+#ifdef ICE_UNALIGNED
+    // Verify inSeq is not aligned and holds the expected values.
+    test(reinterpret_cast<size_t>(inSeq.first) % 8 != 0);
```

Comment:
This should be deterministic and the same on all platforms (CI will prove it!).

The unmarshaling buffer is allocated aligned (on a 8 bytes boundary on a 64-bit system), and then the number of bytes until inSeq.first is always the same.

Reply:
I thought that here we get an unaligned read if `inSeq.first % sizeof(int16_t) != 0`

> Unaligned memory access is the access of data with a size of N number of bytes from an address that is not evenly divisible by the number of bytes N.

https://medium.com/@jkstoyanov/aligned-and-unaligned-memory-access-9b5843b7f4ac

So here if `inSeq.first` is not divisible by `sizeof(int16_t)` we will be reading the sequence values across word boundaries.

Reply:
Fixed